src/components/button-group/button.types.ts

```diff
@@ -0,0 +1,27 @@
+export interface Button {
+    /**
+     * ID of the button. Must be unique.
+     */
+    id: string;
+
+    /**
+     * Text or icon to display in the button.
+     */
+    content: ButtonContent;
```

Comment:
I thought nesting it made it more obvious that you have either text or icon, but it's not really a strong argument since `title` is mandatory (for the icon label) in any case